iphone/Classes/TiUITableView.m

```diff
@@ -2660,32 +2714,34 @@ - (void)didDismissSearchController:(UISearchController *)searchController
   // Since we clear the searchbar, the search string and indexes can be cleared as well.
   [self setSearchString:nil];
   RELEASE_TO_NIL(searchResultIndexes);
+  searchControllerPresenter = nil;
 }
 
 - (void)presentSearchController:(UISearchController *)controller
 {
-  id proxy = [(TiViewProxy *)self.proxy parent];
-  while ([proxy isKindOfClass:[TiViewProxy class]] && ![proxy isKindOfClass:[TiWindowProxy class]]) {
-    proxy = [proxy parent];
-  }
-  UIViewController *viewController = nil;
-  if ([proxy isKindOfClass:[TiWindowProxy class]]) {
-    viewController = [proxy windowHoldingController];
-  } else {
-    viewController = [[TiApp app] controller];
+  if (!searchControllerPresenter) {
+    id proxy = [(TiViewProxy *)self.proxy parent];
+    while ([proxy isKindOfClass:[TiViewProxy class]] && ![proxy isKindOfClass:[TiWindowProxy class]]) {
+      proxy = [proxy parent];
+    }
+    if ([proxy isKindOfClass:[TiWindowProxy class]]) {
+      searchControllerPresenter = [proxy windowHoldingController];
+    } else {
+      searchControllerPresenter = [[TiApp app] controller];
+    }
   }
-  viewController.definesPresentationContext = YES;
+  searchControllerPresenter.definesPresentationContext = YES;
 
-  [viewController presentViewController:controller
-                               animated:NO
-                             completion:^{
-                               CGPoint convertedOrigin = [self.superview convertPoint:self.frame.origin toView:viewController.view];
+  BOOL shouldAnimate = YES;
+  if ([TiUtils isIOS9OrGreater]) {
+    shouldAnimate = NO;
+  }
```

Comment:
Same as before, please refactor the assignment to be less if-based.

Reply:
Done.